File: LeetCode Solutions/October 2022 LeetCoding Challenge/Increasing Triplet Subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first = second = float('inf')
        for i in range(len(nums)):
            if (nums[i] > first):
                second = min(second, nums[i])
            if (nums[i] > second):
                return True
            first = min(first, nums[i])
        return False

File: LeetCode Solutions/LeetCoding Challenge/December 2019 LeetCoding Challenge/Increasing Triplet Subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int left = INT_MAX, mid = INT_MAX;
        for(int i : nums){
            if(i <= left){
                left = i;
            }
            else if(i <= mid){
                mid = i;
            } else{
                return true;
            }
        }
        return false;
    }
};